### PR TITLE
ocamlPackages.wayland: init at 0.2

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lukegb** [comment]
@ofborg build ocamlPackages.wayland

File: pkgs/development/ocaml-modules/wayland/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+, alcotest-lwt
+}:
+
+buildDunePackage rec {
+  pname = "wayland";
+  version = "0.2";
+
+  useDune2 = true;
+
+  src = fetchurl {
```
Comment (lukegb) [inline pkgs/development/ocaml-modules/wayland/default.nix:19]: OOI why this vs fetchFromGitHub?

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/wayland/default.nix:19]: That's a release tarball generated by `dune-release` and uploaded with the release manually (therefore there's `download` in the URL and not `archive`), so the hash should be stable and we can use `fetchurl`. This is pretty common for OCaml dependencies as they often preprocess the source for a release.

Comment (lukegb) [reply pkgs/development/ocaml-modules/wayland/default.nix:19]: Yeah, I saw it was a release artifact, I was just kinda... weirded out by the release source being different from the source source. I guess autotools does weird stuff too, but blergh.

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/wayland/default.nix:19]: Iirc they often have `%%VERSION%%` strings everywhere which get substituted by the release tooling and similar stuff, so yea kinda weird.

Also opam uses these tarballs and I always copy the hash from there. That opam is using those archives is kind of good as it doesn't support post unpack hashing afaik.

File: pkgs/development/ocaml-modules/wayland/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    url = "https://github.com/talex5/ocaml-wayland/releases/download/v${version}/wayland-v${version}.tbz";
+    sha256 = "4eb323e42a8c64e9e49b15a588342bfcc1e99640305cb261d128c75612d9458c";
+  };
+
+  propagatedBuildInputs = [
+    xmlm
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/wayland/default.nix:25]: The `xmlm` library needs not be propagated (only used to build the `wayland-scanner-ocaml` binary program.

**Review by vbgl — CHANGES_REQUESTED** [review]
Please set `minimumOCamlVersion` (to 4.08 at least).

File: pkgs/development/ocaml-modules/wayland/default.nix (hunk 2 above)
Comment (sternenseemann) [reply pkgs/development/ocaml-modules/wayland/default.nix]: Done!

**Review by vbgl — APPROVED** [review]

**Pull request merged**